### PR TITLE
Feat/scheduled events

### LINKED ISSUE #258: Add guild scheduled events endpoints, gateway events, audit log, invite, and permissions changes.
### Summary

Add something that has been documented in ddocs

### What is the feature request for?

The core library

### The Problem

https://github.com/discord/discord-api-docs/commit/1298152172d3359abf069d1b24d4bbc85532b735

### The Ideal Solution

do it

### The Current Solution

not doing it

### Additional Context

https://canary.discord.com/channels/881118111967883295/881124672056467456/909890698873032745

### PR DESCRIPTION
**Pull request by ooliver1**
## Summary

closes #258 

- guild attributes and methods
- intent flag
- dispatch events
- relevant enums and classes
- docs

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

I have tested to some extent but maybe others to test it for different setups would be cool

### CODE REVIEW

**Comment by veni-vidi-code** [comment]
I approve Codeowners change, had no time yet to read rest

File: docs/api.rst
Comment (TAG-Epic) [inline docs/api.rst:2611]: This should probably start with ScheduledEvent or does it not do that in the docs?

File: nextcord/__init__.py
Comment (TAG-Epic) [inline nextcord/__init__.py:7]: We prefer using a second line. See enums file

File: nextcord/enums.py
Comment (TAG-Epic) [inline nextcord/enums.py:604]: Same here?

File: nextcord/ext/commands/__init__.py
Comment (TAG-Epic) [inline nextcord/ext/commands/__init__.py:7]: Same here

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

**Comment by ooliver1** [comment]
1 sec forgot to commit smth

File: nextcord/guild.py
Comment (Skelmis) [inline nextcord/guild.py:3045]: Are these ``Snowflake``? I'm not sure but want to ask

File: nextcord/guild.py
Comment (Skelmis) [inline nextcord/guild.py:3067]: Type `event_id`

File: nextcord/scheduled_events.py
Comment (Skelmis) [inline nextcord/scheduled_events.py:116]: Is there a need for making this 2 lines rather then 1?

File: nextcord/state.py
Comment (Skelmis) [inline nextcord/state.py:1411]: There's a discrepancy between files. Some refer to them as `scheduled_event`, then others are just `guild_event` ..?

**Review by Skelmis — CHANGES_REQUESTED** [review]

File: nextcord/scheduled_events.py
Comment (ooliver1) [reply nextcord/scheduled_events.py:116]: i think it takes up too much line length

File: nextcord/state.py
Comment (ooliver1) [reply nextcord/state.py:1411]: could you show me where its called guild event?

File: nextcord/guild.py
Comment (ooliver1) [reply nextcord/guild.py:3045]: get functions use ints as they search in the dictionary ``Dict[int, obj]``, for consistency with other get_x methods

File: nextcord/state.py
Comment (ooliver1) [reply nextcord/state.py:1411]: edited guild methods

**Review by RPSMain — APPROVED** [review]

**Comment by ooliver1** [comment]
waiting for reviews on this

**Comment by ooliver1** [comment]
user add and remove events added

File: nextcord/client.py
Comment (Skelmis) [inline nextcord/client.py:874]: "The scheduled event's ID to search for." or somethin

File: nextcord/client.py
Comment (Skelmis) [inline nextcord/client.py:1667]: Why isn't this denoted in the docstring guide as returns

```
Returns
--------
List[ScheduledEvent]
 A list of scheduled events
```

File: nextcord/ext/commands/errors.py
Comment (Skelmis) [inline nextcord/ext/commands/errors.py:469]: Is this in the docs?

File: nextcord/guild.py
Comment (Skelmis) [inline nextcord/guild.py:598]:
```Returns
--------
List[ScheduledEvent]
 A list of scheduled events
```

File: nextcord/guild.py
Comment (Skelmis) [inline nextcord/guild.py:3018]: Discuss the default and what it implies

File: nextcord/guild.py
Comment (Skelmis) [inline nextcord/guild.py:3156]: The code denotes that most of these as optional, why not document them as such?

File: nextcord/state.py
Comment (Skelmis) [inline nextcord/state.py:1407]: It's only typed as none, not defaulted to None so this wouldn't occur?

**Review by Skelmis — CHANGES_REQUESTED** [review]

File: nextcord/client.py
Comment (ooliver1) [reply nextcord/client.py:874]: all the other get_x do this, but ill follow this

File: nextcord/client.py
Comment (ooliver1) [reply nextcord/client.py:1667]: properties can be done like this, see persistent_views for example

File: nextcord/ext/commands/errors.py
Comment (ooliver1) [reply nextcord/ext/commands/errors.py:469]: thats a good point

File: nextcord/guild.py
Comment (ooliver1) [reply nextcord/guild.py:598]: also properties can work like this

File: nextcord/guild.py
Comment (ooliver1) [reply nextcord/guild.py:3156]: forgot to update that

File: nextcord/ext/commands/converter.py
Comment (TAG-Epic) [inline nextcord/ext/commands/converter.py:906]: This should be removed as event names are not unique.

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

**Review by Lulalaby — APPROVED** [review]
LGTM Discord API Side.

File: nextcord/ext/commands/converter.py
Comment (ooliver1) [reply nextcord/ext/commands/converter.py:908]: eh other converters like guild use names

Comment (TAG-Epic) [reply nextcord/ext/commands/converter.py:908]: True

**Comment by TAG-Epic** [comment]
Merge conflicts

**Review by vcokltfre — APPROVED** [review]
All seems to be in order, ready to merge!

**Pull request merged**
